core/engine/src/object/jsobject.rs

```diff
@@ -1039,13 +1104,154 @@ impl<T: NativeObject + ?Sized> Debug for JsObject<T> {
+    /// Multiple immutable borrows can be taken out at the same time.
+    ///
+    /// This is the non-panicking variant of [`borrow`](#method.borrow).
+    #[inline]
+    pub fn try_borrow(&self) -> StdResult<Ref<'_, Object<T>>, BorrowError> {
+        self.inner.object.try_borrow().map_err(|_| BorrowError)
+    }
+
+    /// Mutably borrows the object, returning an error if the value is currently borrowed.
+    ///
+    /// The borrow lasts until the returned `GcCellRefMut` exits scope.
+    /// The object be borrowed while this borrow is active.
+    ///
+    /// This is the non-panicking variant of [`borrow_mut`](#method.borrow_mut).
+    #[inline]
+    pub fn try_borrow_mut(&self) -> StdResult<RefMut<'_, Object<T>, Object<T>>, BorrowMutError> {
+        self.inner
+            .object
+            .try_borrow_mut()
+            .map_err(|_| BorrowMutError)
+    }
+
+    /// Checks if the garbage collected memory is the same.
+    #[must_use]
+    #[inline]
+    pub fn equals(lhs: &Self, rhs: &Self) -> bool {
+        Gc::ptr_eq(lhs.inner(), rhs.inner())
+    }
+
+    /// Get the prototype of the object.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the object is currently mutably borrowed.
+    #[inline]
+    #[must_use]
+    #[track_caller]
+    pub fn prototype(&self) -> JsPrototype {
+        self.borrow().prototype()
+    }
+
+    /// Set the prototype of the object.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the object is currently mutably borrowed
+    #[inline]
+    #[track_caller]
+    #[allow(clippy::must_use_candidate)]
+    pub fn set_prototype(&self, prototype: JsPrototype) -> bool {
+        self.borrow_mut().set_prototype(prototype)
+    }
+
+    /// Helper function for property insertion.
+    #[track_caller]
+    pub(crate) fn insert<K, P>(&self, key: K, property: P) -> bool
+    where
+        K: Into<PropertyKey>,
+        P: Into<PropertyDescriptor>,
+    {
+        self.borrow_mut().insert(key, property)
+    }
+
+    /// Inserts a field in the object `properties` without checking if it's writable.
+    ///
+    /// If a field was already in the object with the same name, than `true` is returned
+    /// with that field, otherwise `false` is returned.
+    pub fn insert_property<K, P>(&self, key: K, property: P) -> bool
+    where
+        K: Into<PropertyKey>,
+        P: Into<PropertyDescriptor>,
+    {
+        self.insert(key.into(), property)
+    }
+
+    /// It determines if Object is a callable function with a `[[Call]]` internal method.
+    ///
+    /// More information:
+    /// - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-iscallable
+    #[inline]
+    #[must_use]
+    #[expect(
+        unpredictable_function_pointer_comparisons,
+        reason = "can only use `ptr::fn_addr_eq` on rustc 1.85"
+    )]
+    pub fn is_callable(&self) -> bool {
+        self.inner.vtable.__call__ != ORDINARY_INTERNAL_METHODS.__call__
```

Comment:
nitpick (non-blocking): It doesn't change much but we could use the [`fn_addr_eq()`](https://doc.rust-lang.org/std/ptr/fn.fn_addr_eq.html). (same for `is_constructor()` below)